### PR TITLE
new(pkg/client): handle gRPC unix socket connections

### PR DESCRIPTION
**Pull request by fntlnz**
Signed-off-by: Lorenzo Fontana <lo@linux.com>

**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area client
/area examples

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new: update the gRPC client to support the unix socket
```

Related to: https://github.com/falcosecurity/falco/pull/1217

### CODE REVIEW

**Comment by fntlnz** [comment]
Wait until https://github.com/falcosecurity/falco/pull/1217 is merged

**Review by leodido — DISMISSED** [review]

File: pkg/client/client.go [hunk 1]
```diff
@@ -23,17 +23,36 @@ type Client struct {
...
 func NewForConfig(config *Config) (*Client, error) {
+	if len(config.UnixSocketPath) > 0 {
+		return newUnixSocketClient(config)
+	}
+	return newNetworkClient(config)
+}
+
+func newUnixSocketClient(config *Config) (*Client, error) {
+	dialOption := grpc.WithInsecure()
+	conn, err := grpc.Dial("unix:///var/run/falco.sock", dialOption)
```
Comment (leogr) [inline pkg/client/client.go]:
```suggestion
	conn, err := grpc.Dial(config.UnixSocketPath, dialOption)
```
I guess we wanted the configured value here?

**Review by leogr — COMMENTED** [review]
See the comment.

**Review by leodido — APPROVED** [review]

**Review by leogr — APPROVED** [review]
LGTM

**Pull request merged**